edit/workload/index.vue

```diff
@@ -429,50 +439,28 @@ export default {
     },
 
     async saveService() {
-      const workloadErrors = await this.value.validationErrors(this.value);
+      const { toSave = [], toRemove = [] } = await this.value.servicesFromContainerPorts(this.mode);
 
-      if (!this.createService || workloadErrors.length ) {
+      this.servicesOwned = toSave;
+      this.servicesToRemove = toRemove;
+
+      if (!toSave.length) {
         return;
       }
-      const { ports = [] } = this.container;
-
-      const data = {
-        type: SERVICE,
-        spec: {
-          ports: ports.map((port) => {
-            const name = port.name ? `${ port.name }-${ this.value.metadata.name }` : `${ port.containerPort }${ port.protocol.toLowerCase() }${ port.hostPort || '' }`;
-
-            return {
-              name, protocol: port.protocol, port: port.containerPort, targetPort: port.containerPort
-            };
-          }),
-          selector: this.value.workloadSelector,
-          type:     'ClusterIP'
-        },
-        metadata: {
-          name:        this.value.metadata.name,
-          namespace:   this.value.metadata.namespace,
-          annotations: { [TARGET_WORKLOADS]: `[${ this.value.metadata.namespace }/${ this.value.metadata.name }]` }
-        },
-
-      };
 
-      const service = await this.$store.dispatch(`cluster/create`, data);
-
-      return service.save();
+      return Promise.all(toSave.map(svc => svc.save()));
     },
 
-    async setOwnerRef() {
-      if (!this.createService ) {
+    removeService() {
+      if (!this.servicesToRemove) {
         return;
       }
 
-      const svc = await this.$store.dispatch('cluster/find', {
-        type:     SERVICE,
-        id:   this.value.id
-      });
+      return Promise.all(this.servicesToRemove.map(svc => svc.remove()));
```

Comment:
Can we add an annotation like `management.cattle.io/ui-managed: true` to any services you create and only delete them here if they contain that?  Kind of paranoid about edit workload going through services and deleting them accidentally picking up something unrelated.